src/Parse/Contracts/DataFileInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php namespace Winter\Storm\Parse\Contracts;
+
+interface DataFileInterface
+{
+    /**
+     * Return a new instance of `DataFileInterface` ready for modification of the provided filepath.
+     */
+    public static function open(string $filePath): ?DataFileInterface;
+
+    /**
+     * Set a property within the data.
+     *
+     * @param string|array $key
+     * @param mixed|null $value
+     */
+    public function set($key, $value = null): DataFileInterface;
+
+    /**
+     * Write the current data to a file
+     *
+     * @param string|null $filePath
```

Comment:
@jaxwilko If the parameter type hint indicates the necessary types, there's no need for the docblock to detail the parameter unless you want to add context after it.